rfcs/min-relations.md

```diff
@@ -0,0 +1,273 @@
+        RelationFilters::new()
+            // Note: add_target_filter is additive- i.e. entities this query
+            // matches must be friends with *both* dorothy AND elphaba
+            .add_target_filter::<FriendsWith, _>(dorothy.entity)
+            .add_target_filter::<FriendsWith, _>(elphaba.entity)
+    ).for_each(|mut stress| {
+        println!("{} is friends with Dorothy and Elphaba!", source_entity);
+        stress.val = 1000; 
+    })
+}
+
+/// If we want to combine targets in an "or" fashion, we need a slightly different syntax
+TODO: write example
+
+// Query filters work too!
+fn not_alone(
+    commands: mut Commands, 
+    query: Query<Entity, Without<Relation<FriendsWith>>>,
+    puppy: Res<Puppy>
+) {
+    for lonely_entity in query.iter(){
+        commands.insert_relation(FriendsWith, puppy.entity);
+    }
+}
+
+// So does change detection with Added and Changed!
+fn new_friends(query: Query<&mut Excitement, Added<Relation<FriendsWith>>>){
+    query.for_each(|mut excitement|{excitement.value += 10});
+}
+
+```
+
+TODO: mention relation cleanup on despawn.
+
+Sometimes, we may instead want to look for "entities that are targets of some relation".
+We can do so by iterating over all entities with that relation, and then filtering by target.
+
+```rust
+fn all_children(
+    player_query: Query<Entity, With<Player>>,
+    child_query: Query<&Name, With<Relation<ChildOf>>>,
+) {
+    let player_entity = player_query.single.unwrap();
+    
+    child_query.set_relation_filters(
+        RelationFilters::new()
+            .add_target_filter::<ChildOf, _>(player_entity)
+    );
+
+    for name in child_query.iter() {
+        println!("{} is one of my children.", name) 			
+    }
+}
+```
+
+Finally, you can use the `Entity` returned by your relations to fetch data from the target's components by combining it with `query::get()` or `query::get_component<C>()`.
+
+```rust
+// We need to use a marker component to avoid having multiple mutable references
+fn debts_come_due(
+    mut debt_query: Query<(Entity, Relation<Owes>)>, 
+    mut money_query: Query<&mut Money>,
+) {
+    for (debtor, debt) in debt.query() {
+        for (lender, amount_owed) in debt {
+            *money_query.get(debtor).unwrap()
+                .checked_sub(amount_owed)
+                .expect("Nice kneecaps you got there...");
+            *money_query.get(lender).unwrap() += amount_owed;
+        }
+    }
+}
+```
+
+### Grouping entities
+
+### Entity hierarchies
+
+## Reference-level explanation
+
+TODO: Boxy explains the magic.
+
+## Drawbacks
+
+1. Introduces another core abstraction to users that they will want to learn.
+Rewriting our parent-child code in terms of relations is important, but will introduce the concept to users quite quickly.
+2. Even if relations are more ergonomic than the Entity-in-component approach to accomplish the same tasks, relations will be held to a higher bar for quality.
```

Comment:
Is this a drawback? Easier and higher quality at the same time sounds like a very good thing.

Reply:
Hmmm... It's more work on the devs though! I agree though, this can probably be cut.

Reply:
Cut.